[27.x] Backports and rc1

fanquake opened the pull request:
Backports:
* https://github.com/bitcoin-core/gui/pull/812
* https://github.com/bitcoin-core/gui/pull/813
* https://github.com/bitcoin/bitcoin/pull/30085
* https://github.com/bitcoin/bitcoin/pull/30094
* https://github.com/bitcoin/bitcoin/pull/30097
* https://github.com/bitcoin/bitcoin/pull/30149
* https://github.com/bitcoin/bitcoin/pull/30151

Bump to 27.1rc1.

jonatack committed:
p2p, bugfix: detect addnode cjdns peers in GetAddedNodeInfo()

Addnode (manual) peers connected to us via the cjdns network are currently not
detected by CConnman::GetAddedNodeInfo(), i.e. fConnected is always false.

This causes the following issues:

- RPC `getaddednodeinfo` incorrectly shows them as not connected

- CConnman::ThreadOpenAddedConnections() continually retries to connect them

Github-Pull: #30085
Rebased-From: 684da9707040ce25d95b2954eda50b811136d92c

jonatack committed:
test: add GetAddedNodeInfo() CJDNS regression unit test

Github-Pull: #30085
Rebased-From: d0b047494c28381942c09d0cca45baa323bfcffc

furszy committed:
gui: fix create unsigned transaction fee bump

Github-Pull: https://github.com/bitcoin-core/gui/pull/812
Rebased-From: 671b7a32516d62e1e79393ded4b45910bd08010a

willcl-ark committed:
gui: don't permit port in proxy IP option

Fixes: #809

Previously it was possible through the GUI to enter an IP address:port
into the "Proxy IP" configuration box. After the node was restarted the
errant setting would prevent the node starting back up until manually
removed from settings.json.

Github-Pull: https://github.com/bitcoin-core/gui/pull/813
Rebased-From: 10c5275ba4532fb1bf54057d2f61fc35b51f1e85

willcl-ark committed:
rpc: move UniValue in blockToJSON

Without explicitly declaring the move, these UniValues get copied,
causing increased memory usage. Fix this by explicitly moving the
UniValue objects.

Used by `rest_block` and `getblock` RPC.

Github-Pull: #30094
Rebased-From: b77bad309e92f176f340598eec056eb7bff86f5f

fanquake committed:
doc: update release-notes.md for 27.1

theuni committed:
crypto: disable asan for sha256_sse4 with clang and -O0

Clang is unable to compile the Transform function for that combination of
options.

Github-Pull: #30097
Rebased-From: 141df0a28810470e53fdbc6d32d3cb4020fe3ca1

hebasto reviewed (approved):
ACK 9867e72b995e4a838c8e8e0feb159a9bdee8b76f, I have reproduced backporting locally.

achow101 committed:
windeploy: Renew certificate

Github-Pull: #30149
Rebased-From: 9f4ff1e9659597307f62510f1885ad8da3a1d9a3

hebasto committed:
depends: Fetch miniupnpc sources from an alternative website

The https://miniupnp.tuxfamily.org website is unavailable now.

Github-Pull: #30151
Rebased-From: 21b8a14d37c19ce292d5529597e0d52338db48a9

fanquake committed:
build: bump version to 27.1rc1

fanquake committed:
doc: update manual pages for 27.1rc1

willcl-ark reviewed (approved):
ACK cb6def3855427b613357696ba16a431c7964dbcc

Reviewed that all backports were unmodified, and that associated PR #'s match.

Lightly reviewed release notes and manpage and they look good to me.

stickies-v commented on doc/release-notes.md:
```diff
@@ -1,6 +1,9 @@
+=====================
 
-  <https://bitcoincore.org/bin/bitcoin-core-27.x/>
+Bitcoin Core version 27.1 is now available from:
+
+  <https://bitcoincore.org/bin/bitcoin-core-27.1/>
```
```suggestion
Bitcoin Core version 27.1rc1 is now available from:

  <https://bitcoincore.org/bin/bitcoin-core-27.1/test.rc1/>
```

stickies-v replied:
nit but text is not updated to "271.rc1" in latest force push (link is fine now)

stickies-v reviewed (commented):
ACK cb6def3855427b613357696ba16a431c7964dbcc modulo small release notes fix

All backport commits are clean, except for:
- 77b2321ca03dcbd5f77060510dc8a19e7f4fdfa2 backported from 21b8a14d37c19ce292d5529597e0d52338db48a9: version bumped in https://github.com/bitcoin/bitcoin/pull/29707/ - LGTM

No diff with my local manpage generation.

hebasto committed:
qt: 27.1rc1 translations update

hebasto reviewed (approved):
ACK cb6def3855427b613357696ba16a431c7964dbcc.

Feel free to update translations with `python3 ../bitcoin-maintainer-tools/update-translations.py` or cherry-pick the https://github.com/hebasto/bitcoin/commit/c9c42cc5e9f5740c4700b18a28d809db53880db2 commit.

hebasto reviewed (approved):
re-ACK f30e23e4b35b52728ebaa1a63ec9a6feac51ebb4,

willcl-ark reviewed (approved):
reACK f30e23e4b35b52728ebaa1a63ec9a6feac51ebb4

Didn't review the newly-added individual translations, but apart from that changes LGTM.

stickies-v reviewed (approved):
re-ACK f30e23e4b35b52728ebaa1a63ec9a6feac51ebb4, I'm getting the same translation update results as in 80032d691702f1e5eccb27571066bfc9b2415742

stickies-v commented:
re-ACK 22701a43464ab27ea83c8b49e2732ee647909a10

willcl-ark commented:
reACK 22701a43464ab27ea83c8b49e2732ee647909a10

on range-diff https://github.com/bitcoin/bitcoin/compare/f30e23e4b35b52728ebaa1a63ec9a6feac51ebb4..22701a43464ab27ea83c8b49e2732ee647909a10

hebasto reviewed (approved):
re-ACK 22701a43464ab27ea83c8b49e2732ee647909a10.
